lib/timetable-visualisation/PlanViewer.tsx

```diff
@@ -1,68 +1,127 @@
+
+import { getTimetableEventName } from "@lib/constants";
+import { Tag } from "@lib/gds-components";
+import {
+  getStageProgress,
+  toDefaultLocalDateString,
+} from "@lib/utils/timetable";
 import { DevelopmentPlan, DevelopmentPlanTimetable } from "../types/timetable";
 
 import "govuk-frontend/dist/govuk/govuk-frontend.min.css";
+import { stages } from "./stages";
 
 type PlanViewerProps = {
   developmentPlan: DevelopmentPlan;
   timetableEvents: DevelopmentPlanTimetable[];
 };
 
+type StagePreviewInfo = {
+  name: string;
+  startDate: string;
+  endDate?: string;
+  info?: string;
+};
+
 export const PlanViewer = ({
-  developmentPlan: {
-    name,
-    description,
-    periodStartDate,
-    periodEndDate,
-    developmentPlanType,
-    developmentPlanGeography,
-    organisations,
-    entryDate,
-    startDate,
-    endDate,
-    documentationUrl,
-  },
+  developmentPlan: { name, description },
   timetableEvents,
 }: PlanViewerProps) => {
+  const publishedEvent = timetableEvents.find(
+    (event) =>
+      event.developmentPlanEvent === "local-development-scheme-published"
+  );
+
+  const updatedEvent = timetableEvents.find(
+    (event) => event.developmentPlanEvent === "timetable-updated"
+  );
+
+  if (!publishedEvent) {
+    throw new Error("published event not found");
+  }
+
+  const stagesInfo = useMemo(
+    () => [
+      {
+        name: getTimetableEventName(publishedEvent.developmentPlanEvent),
+        startDate: publishedEvent.eventDate,
+      },
+      ...stages.map<StagePreviewInfo>((stage) => {
+        const startEvent = timetableEvents.find(
+          (event) => event.developmentPlanEvent === stage.startEventKey
+        );
+
+        const endEvent = timetableEvents.find(
+          (event) => event.developmentPlanEvent === stage.endEventKey
+        );
+
+        if (!startEvent) {
+          throw new Error("event not found");
+        }
+        return {
+          name: stage.title,
+          startDate: startEvent.eventDate,
+          endDate: endEvent?.eventDate,
+          info: startEvent.notes,
+        };
+      }),
+    ],
+    [publishedEvent, timetableEvents]
+  );
+
   return (
-    <div data-testid="plan-viewer">
-      <div>
-        <h2 className="govuk-heading-l">{name}</h2>
-        <p className="govuk-body-l">{description}</p>
-      </div>
-      <div className="govuk-body">
-        <p>{`Period Start: ${periodStartDate}`}</p>
-        <p>{`Period End: ${periodEndDate}`}</p>
-        <p>{`Plan Type: ${developmentPlanType}`}</p>
-        <p>{`Plan Geography: ${developmentPlanGeography}`}</p>
-        <p>{`Organisations: ${organisations}`}</p>
-        <p>{`Entry Date ${entryDate}`}</p>
-        <p>{`Start Date: ${startDate}`}</p>
-        <p>{`End Date: ${endDate}`}</p>
-        {documentationUrl && <a href={documentationUrl}>Documentation</a>}
-      </div>
+    <div className="govuk-body" data-testid="plan-viewer">
+      <h2 className="govuk-heading-l">{name}</h2>
```

Comment:
We needs to append "timetable" to the plan name:

```suggestion
      <h2 className="govuk-heading-l">{name} timetable</h2>
```

Reply:
done :)